content/blog/migrate-kubedns-to-coredns-in-kubernetes.md

```diff
@@ -0,0 +1,117 @@
+This means that CoreDNS will be available as the standard in Kubernetes via the installation toolkits such as kubeadm, kube-up, minikube and kops.
+
+This document will guide you to migrating the DNS service from CoreDNS to kube-dns when using the various tools available to spin up a Kubernetes cluster.
+
+## Installing CoreDNS via Kubeadm
+
+There is an extensive guide on how to install CoreDNS instead of kube-dns via Kubeadm available [here](https://coredns.io/2018/01/29/deploying-kubernetes-with-coredns-using-kubeadm).
+Currently, from Kubernetes v1.10, CoreDNS supports the translation of the kube-dns configmap to CoreDNS configmap.
+That is, if you had configured `stubdomains`, `upstreamnameservers` and `federation` via the kube-dns configMap, it will now be translated automatically to the equivalent CoreDNS configmap during when choosing to install CoreDNS using `kubeadm upgrade`.
+
+`Stubdomain` and `upstreamnameserver` in kube-dns translates to the [`proxy`](https://coredns.io/plugins/proxy/) in CoreDNS.
+The `federation` in kube-dns has an equivalent [`federation`](https://coredns.io/plugins/federation/) in CoreDNS.
+
+
+Example ConfigMap of kube-dns.
+~~~text
+apiVersion: v1
+data:
+  federations: |
+    {"foo" : "foo.feddomain.com"}
+  stubDomains: |
+    {"abc.com" : ["1.2.3.4"], "my.cluster.local" : ["2.3.4.5"]}
+  upstreamNameservers: |
+    ["8.8.8.8", "8.8.4.4"]
+kind: ConfigMap
+metadata:
+  creationTimestamp: 2018-01-22T20:21:56Z
+  name: kube-dns
+  namespace: kube-system
+~~~
+
+CoreDNS Corefile after translation.
+
+~~~text
+   .:53 {
+        errors
+        health
+        kubernetes cluster.local  in-addr.arpa ip6.arpa {
+           upstream  8.8.8.8 8.8.4.4
+           pods insecure
+           fallthrough in-addr.arpa ip6.arpa
+        }
+       federation cluster.local {
+           foo foo.feddomain.com
+        }
+        prometheus :9153
+        proxy .  8.8.8.8 8.8.4.4
+        cache 30
+    }
+    abc.com:53 {
+        errors
+        cache 30
+        proxy . 1.2.3.4
+    }
+    my.cluster.local:53 {
+        errors
+        cache 30
+        proxy . 2.3.4.5
+    }
+~~~
+
+
+
+## Installing CoreDNS via Minikube.
+
+CoreDNS is available in the `addon manager` disabled by default.
+
+~~~text
+$ minikube addons list
+- kube-dns: enabled
+- registry: disabled
+- registry-creds: disabled
+- freshpod: disabled
+- addon-manager: enabled
+- dashboard: enabled
+- coredns: disabled
+- heapster: disabled
+- efk: disabled
+- ingress: disabled
+- default-storageclass: enabled
+- storage-provisioner: enabled
+~~~
+
+To enable CoreDNS, run the following command:
+> NOTE: Ensure to disable kube-dns after enabling CoreDNS.
+
+~~~
+$ minikube addons enable coredns
+coredns was successfully enabled
+~~~
+
+
+## CoreDNS in kube-up
+
+[Kube-up](https://kubernetes.io/docs/getting-started-guides/scratch/) is another way to start a Kubernetes cluster, now mostly used for deploying Kubernetes in GCE for end-to-end (e2e) testing purposes.
```

Comment:
Maybe we should inform here about the other variable to have DNS started. But sure that the reader understand that both of these variables need to he TRUE to have CoreDNS installed.